components/tidb_query_expr/src/impl_time.rs

```diff
@@ -700,6 +700,38 @@ pub fn add_duration_and_string(
     Ok(Some(res))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn duration_duration_time_diff(
+    ctx: &mut EvalContext,
+    arg1: &Duration,
+    arg2: &Duration,
+) -> Result<Option<Duration>> {
+    let res = match arg1.checked_sub(*arg2) {
+        Some(res) => res,
+        // `check_sub` returns `None` if the sub operation overflow/underflow i64 bound or mysql_time_value bound.
+        // and we need to treat these two case separately.
+        // if `arg1 - arg2` is in [`MAX_NANOS`, `i64::MAX`], return max value of mysql `TIME` type.
+        // if `arg1 - arg2` is in [`i64::MIN`, `-MAX_NANOS`], return min value of mysql `TIME` type.
+        // if `arg1 - arg2` is overflow or underflow i64, return `None`.
+        None if !arg1.is_neg() && i64::MAX - arg1.to_nanos() > -arg2.to_nanos() => {
```

Comment:
Maybe `arg1.to_nanos() - arg2.to_nanos() < i64::MAX` is more readable?

Reply:
Reaching the `None` case means sub operation `arg1.to_nanos() - arg2.to_nanos()` may cause overflow/underflow. Thus, I transform the inequality to make sure the lhs is valid. Perhaps `arg1.to_nanos() - i64::MAX <= arg2.to_nanos()` is a better option here?

Reply:
Yes, the second style is more readable for me.